### ISSUE TITLE
Issue installing 0.8.1 on Windows/GitHub Actions

### ISSUE THREAD

**Issue by davidmezzetti** [duckdb/duckdb#7933]
### What happens?

Attempting to install duckdb==0.8.1 on Windows via GitHub actions fails.

```
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(267): error C2485: '__restrict': unrecognized extended attribute
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(269): error C2485: '__restrict': unrecognized extended attribute
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(273): error C2485: '__restrict': unrecognized extended attribute
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(275): error C2485: '__restrict': unrecognized extended attribute
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(279): error C2485: '__restrict': unrecognized extended attribute
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(281): error C2485: '__restrict': unrecognized extended attribute
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/internal/jemalloc_internal_macros.h(36): warning C4005: 'restrict': macro redefinition
  duckdb/extension/jemalloc/jemalloc/include\jemalloc/internal/jemalloc_internal_decls.h(87): note: see previous definition of 'restrict'
  error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.35.32215\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
  [end of output]
```

Build failure example: https://github.com/neuml/txtai/actions/runs/5257654781/jobs/9500968762#step:9:557

### To Reproduce

Create a simple GitHub Action script to `pip install duckdb==0.8.1` on Windows. This issue does not occur with `duckdb==0.8.0`. 

### OS:

Windows

### DuckDB Version:

0.8.1

### DuckDB Client:

Python

### Full Name:

David Mezzetti

### Affiliation:

NeuML

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mause** [comment]
I will say though that jemalloc should only be enabled on Linux, so we probably need to fix our setup.py script

**Comment by Mause** [comment]
Looking at https://github.com/duckdb/duckdb/blob/5c94293b1302d9a6f01f81f3c6aeaa0f4ef3fb78/tools/pythonpkg/setup.py#L67 that is still the case, so more investigation is needed

**Comment by carlopi** [comment]
@Mause beat me, if you retry the job I would expect it to work since CI has since completed.

I think the other issue is wider, and it's that `__restrict` keyword is not recognised by Visual studio code, so a potential workaround is using a pattern like the one used in `third_party/fsst/fsst.h`:
```c++
#ifdef _MSC_VER
#define __restrict
#endif
```

**Comment by Mytherin** [comment]
Should be fixed now

**Comment by Mause** [comment]
> Should be fixed now

Where was the jemalloc issue fixed sorry?

**Comment by Mytherin** [comment]
That was not fixed but the installation on Windows should be fixed. We could leave this open to track the jemalloc issue or create a separate issue

**Comment by mmaitre314** [comment]
Would you have more details about what the fix looked like? Pip installing `duckdb==0.9.2` on my Windows dev machine hit a similar issue, so I am trying narrow down where the installation went wrong.

Pip output: [pip.log](https://github.com/duckdb/duckdb/files/13772901/pip.log)

```
arena.cpp
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(267): error C2485: '__restrict': unrecognized extended attribute
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(269): error C2485: '__restrict': unrecognized extended attribute
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(273): error C2485: '__restrict': unrecognized extended attribute
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(275): error C2485: '__restrict': unrecognized extended attribute
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(279): error C2485: '__restrict': unrecognized extended attribute
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/jemalloc.h(281): error C2485: '__restrict': unrecognized extended attribute
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/internal/jemalloc_internal_macros.h(36): warning C4005: 'restrict': macro redefinition
duckdb_build/extension/jemalloc/jemalloc/include\jemalloc/internal/jemalloc_internal_decls.h(87): note: see previous definition of 'restrict'
```

**Comment by mmaitre314** [comment]
Thanks for the prompt reply! Switching to Python 3.11 fixed the issue.
